spring-cloud-data-rest-client/src/main/java/org/springframework/cloud/data/rest/client/CloudDataTemplate.java

```diff
@@ -55,10 +60,16 @@ public CloudDataTemplate(URI baseURI) {
 		resources.put("streams/definitions", new UriTemplate(resourceSupport.getLink("streams").getHref() + "/definitions"));
 		resources.put("streams/deployments", new UriTemplate(resourceSupport.getLink("streams").getHref() + "/deployments"));
 		this.streamTemplate = new StreamTemplate(restTemplate, resources);
+		this.counterTemplate = new CounterTemplate(restTemplate, resourceSupport);
```

Comment:
Can we reduce that knowledge on the CloudDataTemplate side to be based on prefixes, such that it hands out all links with a given prefix to a specific template? That might be a nice compromise. That said, since CloudDataController is the "master" controller, I think it's better to err on the side of *it* having more knowledge than to pass all of the links to the specific controllers, thus giving them visibility to each other's links.